### PR TITLE
Add new TrackCommand method with Dictionary<string, object> argument

### PR DESCRIPTION
**Pull request by pinzart90**
Task:
DYN-5490

Add new Track method
Refactor old one because of ambiguous signature (ok to change because it was introduced in 3.0) 

needs this PR https://git.autodesk.com/Dynamo/Analytics.NET/pull/73

### CODE REVIEW

File: src/DynamoCore/Logging/DynamoAnalyticsClient.cs [hunk 1]
```diff
@@ -357,26 +358,52 @@ public Task<IDisposable> CreateTaskTimedEvent(Categories category, string variab
...
+                    }
+                }    
+
                 e.Track();
                 return e;
             }
         }
 
-        public Task<IDisposable> CreateTaskCommandEvent(string name, string description, int? value)
+        public Task<IDisposable> CreateTaskCommandEvent(string name, string description, int value)
```
Comment (mjkkirschner) [inline src/DynamoCore/Logging/DynamoAnalyticsClient.cs:393]: I am likely missing something but - why doesn't one of these methods exist that expects both a value and attributes dictionary?

Comment (pinzart90) [reply src/DynamoCore/Logging/DynamoAnalyticsClient.cs:393]: Argggg, part of the interface...

Comment (mjkkirschner) [reply src/DynamoCore/Logging/DynamoAnalyticsClient.cs:393]: It's okay to break interfaces now though - right?

File: src/NodeServices/IAnalyticsClient.cs [hunk 2]
```diff
@@ -559,7 +560,17 @@ public interface IAnalyticsClient
         /// <param name="description">Event description</param>
         /// <param name="value">A metric value associated with the event</param>
         /// <returns>Event as IDisposable</returns>
-        Task<IDisposable> CreateTaskCommandEvent(string name, string description, int? value);
```
Comment (pinzart90) [inline src/NodeServices/IAnalyticsClient.cs:562]: this was introduced 3 weeeks ago (for 3.0).
I think breaking the interface should still be ok...

File: src/NodeServices/IAnalyticsClient.cs [hunk 3]
```diff
@@ -559,7 +560,17 @@ public interface IAnalyticsClient
...
+
+        /// <summary>
+        /// Creates a new task command event of the given name. Start of the 
+        /// command is tracked. When the task is completed and the event is disposed, it's completion is tracked.
+        /// </summary>
+        /// <param name="name">Command name</param>
+        /// <param name="description">Event description</param>
+        /// <param name="parameters">A dictionary of (stringl, object) associated with the event</param>
+        /// <returns>Event as IDisposable</returns>
+        Task<IDisposable> CreateTaskCommandEvent(string name, string description, IDictionary<string, object> parameters);
```
Comment (pinzart90) [inline src/NodeServices/IAnalyticsClient.cs:573]: this will break it regardless

File: src/DynamoCore/Logging/DynamoAnalyticsClient.cs (hunk 1 above)
Comment (pinzart90) [reply src/DynamoCore/Logging/DynamoAnalyticsClient.cs:393]: we broke it 3 weeks ago 
I hope it is ok now too

**Comment by pinzart90** [comment]
I find it very difficult just to add a new tracking API.
Interface, Implementing class and static wrapper - all public APIs. and the interface means breaking changes when we add new methods...

**Comment by mjkkirschner** [comment]
@pinzart90 should we just drop the interface?

**Comment by pinzart90** [comment]
> @pinzart90 should we just drop the interface?

Well, it is used to isolate analytics types between DynamoCore and DynamoServices.
If we find a better way to isolate them without the use of an Interface...then I am all for it

**Comment by BogdanZavu** [comment]
Since we are breaking the API can we just have a single CreateTaskCommandEvent & CreateCommandEvent with all 4 parameters ?

**Comment by BogdanZavu** [comment]
I assume we still need to add a ref to the new Analytics.net package unless you want a new PR for that.
LGTM!

**Review by BogdanZavu — APPROVED** [review]

**Comment by BogdanZavu** [comment]
Looks like we are good on this too ?

**Comment by BogdanZavu** [comment]
@pinzart90 @mjkkirschner this is working as expected for us ( I'm seeing the key-value pairs in the adp log ).
Can we please cherry-pick this in 3.0 and also publish some packages after that ? Thanks.

**Pull request merged**